app/helpers/sessions_helper.rb

```diff
@@ -1,9 +1,7 @@
 module SessionsHelper
 
-  def show_flash
-    if flash[:alert]
-      content_tag :p, flash[:alert], class: 'flash alert'
-    end
+  def show_flash(key)
+    content_tag :p, flash[:"#{key}"], class: "flash #{key}"
```

Comment:
```suggestion
    content_tag :p, flash[key], class: "flash #{key}"
```
у тебя вызов хелпера выглядит так: `show_flash(:alert)`, т.е. в аргументах ты передаешь символ
к объекту `flash` можно обращаться с использованием символа, как ключа
и именно символ будет в переменной `key`
дополнительные преобразования не нужны

Reply:
в старом варианте была проверка `if flash[:alert]`
она вполне логична
в объекте `flash` может не быть содержимого под ключом пришедшим в аргументах
в текущем коде также будет нелишним проверить есть ли флеш с пришедшим ключом в наличии